### DISCUSSION TITLE
TooManyRequests error

### DISCUSSION THREAD

**Discussion by sergioteula** [sergioteula/python-amazon-paapi#59] [Help] [answered]
If you're getting this error it's probably due to some of below reasons:

- You're doing more requests than you're allowed to do by Amazon. Modify the throttling parameter to limit requests.
- Your API account is not allowed to make requests to the target region. Try changing the region.
- Your API credentials are being used by another service, so maybe requests limit is being reached. Stop all services using the API and try again.

There are some interesting resources you can check to try finding the root cause:

- [Amazon API rate limits](https://webservices.amazon.com/paapi5/documentation/troubleshooting/api-rates.html)
- [Product Advertising API 5.0 Scratchpad](https://webservices.amazon.com/paapi5/scratchpad/index.html)
- [Closed issue about this error](https://github.com/sergioteula/python-amazon-paapi/issues/32)

You can ask for help here if you need, but please don't open new issues.

**Comment by barbax7** [comment]
I'd like to handle the AmazonException in case of the TooManyRequests error, to avoid crashing the script. It's common for me that I execute more than one worker that could send requests at the same time.

What have I to write in try/except block so?

**Reply by aashishsaini75** [reply]
@barbax7 , You can call the API and retry the same until the "Too Many Requests" error gets away, The API will get successfully called as soon as it will get the bandwidth from other crawlers, like ...

`
while True:
                    try:
                        time.sleep(1)
                        product = amazon.get_product(asin)
                        price = product.prices.price.value
                    except Exception as E:
                        if "Too Many Requests" in str(E):
                                continue
                    break`

**Reply by sergioteula** [reply]
I will add better exceptions on next API version so you will be able to handle them more efficiently.

**Comment by sergioteula** [comment, marked as answer]
You can now handle ToManyRequests exception with the new API version.

**Reply by barbax7** [reply]
Yes, very useful, thank you!

**Comment by gjaon** [comment]
This is 2024, and I'm still encountering the same error. I used the code in the github docs for node.js. Is there something I need to know?

**Reply by barbax7** [reply]
The error doesn't depend on the library, but on your amazon account (as said in first topic).

**Reply by gjaon** [reply]
Okay, thank you.
